### DISCUSSION TITLE
🚀 Feature request - Allow unique 'required field' option for fields in Publish Layouts per channel.

### DISCUSSION THREAD

**Discussion by ttmwebdesign** [ExpressionEngine/ExpressionEngine#4173] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I have created a new channel that is reusing an existing field that is *not* required in other channels, but I would like it to be in this new one. However, I cannot do that without affecting existing channels & entries where the field is present, as well as future entries where this field is used (but is optional at this time).

**Describe the solution you'd like**
If possible, in a channels Publish Layout options, in addition to the 'hide' and 'collapse' options, I would like to see an override choice to be able to make the field required per-channel vs. in all channels.

Thanks.

**Comment by intoeetive** [comment]
Adding my thoughts for future:
- we might want to review how "required" flag is set as a whole. Currently it's per-field, and it could be more useful if we make it more aware of context. Example: required per channel, or required when used in certain Fluid field. We could use the UI (and backend approach) similar to what we have when we assign category groups to the channel.
